src/components/select/select.spec.js

```diff
@@ -333,6 +333,78 @@ describe('<md-select>', function() {
 
     }));
 
+    describe('md-select-header behavior', function() {
+      it('supports rendering a md-select-header', inject(function($rootScope, $compile) {
+        $rootScope.val = [1];
+        var select = $compile('<md-input-container>' +
+                                '<label>Label</label>' +
+                                '<md-select multiple ng-model="val" placeholder="Hello World">' +
+                                  '<md-select-header class="demo-select-header">' +
+                                    '<span>Hello World</span>' +
+                                  '</md-select-header>' +
+                                  '<md-optgroup label="stuff">' +
+                                     '<md-option value="1">One</md-option>' +
+                                     '<md-option value="2">Two</md-option>' +
+                                     '<md-option value="3">Three</md-option>' +
+                                  '</md-optgroup>' +
+                                '</md-select>' +
+                              '</md-input-container>')($rootScope);
+
+        var header = select.find('md-select-header');
+        var headerContent = header.find('span');
+
+        expect(headerContent.text()).toBe('Hello World');
+      }));
+
+      it('does not render the label in md-optgroup if md-select-header is present', inject(function($rootScope, $compile) {
+        $rootScope.val = [1];
+        var select = $compile('<md-input-container>' +
+                                '<label>Label</label>' +
+                                '<md-select multiple ng-model="val" placeholder="Hello World">' +
+                                  '<md-select-header class="demo-select-header">' +
+                                    '<span>Hello World</span>' +
+                                  '</md-select-header>' +
+                                  '<md-optgroup label="stuff">' +
+                                    '<md-option value="1">One</md-option>' +
+                                    '<md-option value="2">Two</md-option>' +
+                                    '<md-option value="3">Three</md-option>' +
+                                  '</md-optgroup>' +
+                                '</md-select>' +
+                              '</md-input-container>')($rootScope);
+
+        var optgroupLabel = select[0].querySelector('._md-container-ignore');
+
+        expect(optgroupLabel).toBe(null);
+      }));
+    });
+
+    it('does not allow keydown events to propagate from inside the md-select-menu', inject(function($rootScope, $compile) {
```

Comment:
Ohh aha, for the sake of the new component (where inputs can be put in the `md-select-header`) I had to override this logic and wrote a test to ensure that.
 
I then noticed there was no test to cover the original keydown logic (which was a larger bit of code) and thought I might add this as a sort of parallel 'negative' to the test I added.

I would be okay with removing this, it was essentially just a copy of the other test I wrote with `not` added to the expects. I will leave it to your discretion :)